sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -159,6 +161,13 @@ class SessionResourceLoader(session: SparkSession) extends FunctionResourceLoade
     }
   }
 
+  protected def resolveJars(path: String): List[String] = {
+    new Path(path).toUri.getScheme match {
```

Comment:
Shouldn't we use `URI.create(path)` a single time, then re-use the URI in this line and the one below? I also think I remember you mentioning elsewhere that `new Path(path).toUri` can lose information.

Reply:
Similar comment for `addJar` below.

Reply:
Updated